### ISSUE TITLE
"Server communication error" even if delete request was successful 

### ISSUE THREAD

**Issue by anjeylink** [api-platform/admin#440]
**Admin version**: 2.8.3

**Description**  
I get "Server communication error" every time I delete a record.

**Possible solution**  
All delete requests done by `dataProvider.delete()` got rejected by `fetchJsonLd` because of changes done in this [commit](https://github.com/api-platform/api-doc-parser/commit/a20518e36c0431eb1cfa51ed9a17ff9b980dbe4f#diff-3e15d96c3d25a2ff99bb0c02ead6b4ee10d9a441c410c1cc09d37976546c81d8R26)

**Comment by gartner** [comment]
I see the same behavior in Admin 3.0.0-rc

**Comment by theMadness** [comment]
My current workaround is to decorate the dataProvider

```
  return {
    ...dataProvider,
    // @ts-ignore
    delete: (
      resource: string,
      params: ApiPlatformAdminDeleteParams,
    ): Promise<DeleteResult<ApiPlatformAdminRecord>> => {
      const response = dataProvider.delete(resource, params).then(
        result => {
          console.log({ result });

          return result;
        },
        error => {
          if (error.response.status === 204) {
            return { data: params.previousData };
          }

          return Promise.reject(error);
        },
      );
      console.log(response);

      return response;
    },
  };
```

(also accepting suggestions on how to get rid of the typescript issues that force me to use @ts-ignore. :P

**Comment by alanpoulain** [comment]
It should be solved in 2.8.4.
